tapdb/assets_common.go

```diff
@@ -84,21 +84,55 @@ type UpsertAssetStore interface {
 		error)
 }
 
+var (
+	// ErrEncodeGenesisPoint is returned when encoding a genesis point
+	// before upserting it fails.
+	ErrEncodeGenesisPoint = errors.New("unable to encode genesis point")
+
+	// ErrReadOutpoint is returned when decoding an outpoint fails.
+	ErrReadOutpoint = errors.New("unable to read outpoint")
+
+	// ErrUpsertGenesisPoint is returned when upserting a genesis point
+	// fails.
+	ErrUpsertGenesisPoint = errors.New("unable to upsert genesis point")
+
+	// ErrUpsertGroupKey is returned when upserting a group key fails.
+	ErrUpsertGroupKey = errors.New("unable to upsert group key")
+
+	// ErrUpsertInternalKey is returned when upserting an internal key
+	// fails.
+	ErrUpsertInternalKey = errors.New("unable to upsert internal key")
+
+	// ErrUpsertScriptKey is returned when upserting a script key fails.
+	ErrUpsertScriptKey = errors.New("unable to upsert script key")
+
+	// ErrNoAssetGroup is returned when no matching asset group is found.
+	ErrNoAssetGroup = errors.New("no matching asset group")
+
+	// ErrGroupGenesisInfo is returned when fetching genesis info for an
+	// asset group fails.
+	ErrGroupGenesisInfo = errors.New("unable to fetch group genesis info")
+
+	// ErrTapscriptRootSize is returned when the given tapscript root is not
+	// exactly 32 bytes.
+	ErrTapscriptRootSize = errors.New("tapscript root invalid: wrong size")
+)
+
 // upsertGenesis imports a new genesis point into the database or returns the
 // existing ID if that point already exists.
 func upsertGenesisPoint(ctx context.Context, q UpsertAssetStore,
 	genesisOutpoint wire.OutPoint) (int64, error) {
 
 	genesisPoint, err := encodeOutpoint(genesisOutpoint)
 	if err != nil {
-		return 0, fmt.Errorf("unable to encode genesis point: %w", err)
+		return 0, fmt.Errorf("%w: %w", ErrEncodeGenesisPoint, err)
```

Comment:
I know newer Golang versions allow for double wrapping errors. But not sure how that works exactly? Do we really want to do that?

Reply:
Good catch, I tested this in the playground and what I added doesn't work as expected. I mostly wanted to dedupe these error messages; AFAICT callers right now aren't inspecting the concrete error type.

So I think the fix would be to just define these new variables as strings instead of errors, and use:

`fmt.Errorf("%s: %w)`.

Callers would then have to use `strings.Contains()` instead of `errors.Is()`.